src/coreclr/src/binder/assemblybinder.cpp

```diff
@@ -1341,14 +1345,16 @@ HRESULT AssemblyBinder::BindUsingPEImage(/* in */  ApplicationContext *pApplicat
                 // load.
                 IF_FAIL_GO(bindResult.GetAsAssembly()->GetMVID(&boundMVID));
```

Comment:
Might be a good idea to `tracer.SetFoundAssembly(bindResult.GetAsAssembly())` right after the `HaveResult` check. This is so that if any of the calls above fail (`IF_FAIL_GO`) the tracer will have an assembly to report the HR related to.